### PR TITLE
HTTPCookie is a subtype of HTTPClient

### PR DESCRIPTION
**Pull request by Yasumoto**
This prevents it from colliding with Foundation's top-level type, as described in #32.

I think putting it in `HTTPHandler` matches it with `Request`, `Response,` and `Body`, but I also considered keeping it in its own file as well as in `SwiftNIOHTTP.swift`. Ideas welcome!

### CODE REVIEW

File: Sources/NIOHTTPClient/HTTPHandler.swift [hunk 1]
```diff
@@ -94,6 +94,111 @@ public extension HTTPClient {
         public var headers: HTTPHeaders
         public var body: ByteBuffer?
     }
+
+    struct HTTPCookie {
```
Comment (artemredkin) [inline Sources/NIOHTTPClient/HTTPHandler.swift]: I think having in in a separate file would be better, we can have `extension HTTPClient` there, that will prevent `HTTPHandler.swift` from becoming overloaded.

Comment (Yasumoto) [reply Sources/NIOHTTPClient/HTTPHandler.swift]: Cool, works for me. Think it should be called something like `HTTPClient+HTTPCookie.swift`?

Comment (artemredkin) [reply Sources/NIOHTTPClient/HTTPHandler.swift]: Yep, that will work!

Comment (Yasumoto) [reply Sources/NIOHTTPClient/HTTPHandler.swift]: Lemme know what you think! 🙇‍♂️

File: Sources/NIOHTTPClient/HTTPClient+HTTPCookie.swift [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import NIOHTTP1
+
+extension HTTPClient {
+    public struct HTTPCookie {
```
Comment (tomerd) [inline Sources/NIOHTTPClient/HTTPClient+HTTPCookie.swift]: might as well name it `Cookie` under the new structure

Comment (Yasumoto) [reply Sources/NIOHTTPClient/HTTPClient+HTTPCookie.swift]: Good call, fixed!

**Comment by artemredkin** [comment]
@swift-server-bot test this please

**Pull request merged**